### PR TITLE
describegpt: add --user-agent in help message

### LINKED ISSUE #1082: `describegpt`: add --user-agent option
Hi @rzmk , now that we have an MVP implementation (#896 ), let's use the issue tracker to collaborate on releasing `describegpt`.

I'll create an issue for each feature we need for release.

The first one, is as the title says.  Refer to `fetch` for an example.

### PR DESCRIPTION
**Pull request by rzmk**
## 🗺 Overview

Resolves #1082. Adds the `--user-agent` option to `describegpt`.

### CODE REVIEW

**Comment by rzmk** [comment]
I'm curious how to verify this works properly.

File: src/cmd/describegpt.rs [hunk 1]
```diff
@@ -42,15 +45,28 @@ struct Args {
...
 const MODEL: &str = "gpt-3.5-turbo-16k";
 
-fn get_completion(api_key: &str, messages: &serde_json::Value, max_tokens: Option<i32>) -> String {
+fn get_completion(
+    api_key: &str,
+    messages: &serde_json::Value,
+    max_tokens: Option<i32>,
+    flag_user_agent: Option<String>,
+) -> String {
+    // If --user-agent is not specified, use an empty string as the default value
```
Comment (jqnatividad) [inline src/cmd/describegpt.rs:60]: Hi @rzmk , you don't need to do this. Just set user_agent when setting up the reqwest client...

File: src/cmd/describegpt.rs [hunk 2]
```diff
@@ -42,15 +45,28 @@ struct Args {
...
+    let user_agent = flag_user_agent.unwrap_or_default();
+
     // Create client with timeout
     let timeout_duration = Duration::from_secs(60);
-    let client = Client::builder().timeout(timeout_duration).build().unwrap();
+    let client = Client::builder()
+        .timeout(timeout_duration)
+        .user_agent(user_agent.clone())
```
Comment (jqnatividad) [inline src/cmd/describegpt.rs]: Just set user_agent right here, replacing this:

```rust
.user_agent(user_agent.clone())
```

with

```rust
.user_agent(util::set_user_agent(args.flag_user_agent)?)
```

**Review by jqnatividad — CHANGES_REQUESTED** [review]
Hi @rzmk , just a few tweaks - see how its done by the `fetch` command for a quick reference

File: src/cmd/describegpt.rs [hunk 3]
```diff
@@ -42,15 +45,28 @@ struct Args {
...
-fn get_completion(api_key: &str, messages: &serde_json::Value, max_tokens: Option<i32>) -> String {
+fn get_completion(
+    api_key: &str,
+    messages: &serde_json::Value,
+    max_tokens: Option<i32>,
+    flag_user_agent: Option<String>,
+) -> String {
+    // If --user-agent is not specified, use an empty string as the default value
+    let user_agent = flag_user_agent.unwrap_or_default();
+
```
Comment (jqnatividad) [inline src/cmd/describegpt.rs]: Hi @rzmk , you don't need to do user_agent var. Just set user_agent directly when setting up the reqwest client...

**Review by jqnatividad — CHANGES_REQUESTED** [review]
Made requested change on unneeded user_agent var clearer...

**Comment by rzmk** [comment]
`args.flag_user_agent` wasn't in scope of the function `get_completion`, same with `args.max_token`. I've passed `args` now and removed the `user_agent` variable.

**Pull request merged**